ca/certificates.go

```diff
@@ -579,6 +562,79 @@ func ValidateCertChain(rootPool *x509.CertPool, certs []byte, allowExpired bool)
 	return parsedCerts, nil
 }
 
+// newLocalSigner validates the signing cert and signing key to create a local signer, which accepts a crypto signer and a cert
+func newLocalSigner(keyBytes, certBytes []byte, certExpiry time.Duration, rootPool, intermediatePool *x509.CertPool) (*LocalSigner, error) {
+	switch {
+	case len(keyBytes) == 0 && len(certBytes) == 0: // not valid signer, but no error
+		return nil, nil
+	case len(keyBytes) > 0 && len(certBytes) > 0: // possibly valid signer
+	default:
+		return nil, errors.New("must provide both a signing key and a signing cert, or neither")
+	}
```

Comment:
You could move this to a separate function like `possibleValidSigner` if you expect to have multiple call sites for `newLocalSigner` in the future.